### ISSUE TITLE
sample_shape in non extended model

### ISSUE THREAD

**Issue by rcabanasdepaz** [PGM-Lab/InferPy#115]
Which should be the default sample_shape in the non-extended model? 1 or empytset?

```
@models.probmodel
def model():
    with models.datamodel():
        x = models.Normal(np.zeros([2,4], np.float32), 1., name='x')
```

x.shape == [1,2,4] or x.shape == [2,4]. If we consider that the sample_shape is at least 1, any variable inside a data model will always have a shape of [N,....]. We assume that in the non-expanded model N==1.

**Comment by rcabanasdepaz** [comment]
A problem if we consider N==[], is that the following code will not work:

```

k, d = 5, 10


@models.probmodel
def pca():
    w = models.Normal(np.zeros([k,d]), 1, name="w")

    with models.datamodel():
        z = models.Normal(np.zeros(k), 1, name="z")
        x = models.Normal(tf.matmul(z,w), 1, name="x")




m = pca()



```

**Comment by jcozar87** [comment]
As @rcabanasdepaz exposed, operations are dependent on the tensor shapes. In order to avoid weird behaviours, we will use `sample_shape=N` for random variables inside _datamodel_ (which is instantiated dynamically to a particular value). By default, N=1.

`sample_shape` must be an `integer`; otherwise operations might fail: i.e.: consider the operation `tf.tensordor(a, b, axis=[...])`, depending on the `sample_shape` axis indices must be different.

Consider that if you have the following code, as sample_shape is used in _datamodel_, users must use `tf.tensordot` instead of `tf.matmul`:

```python
@models.probmodel
def model():
    w = models.Normal(np.zeros([4,2], np.float32), 1., name='w')  # shape = [4, 2]
    with models.datamodel():
        x = models.Normal(np.zeros([2,4], np.float32), 1., name='x')  # shape = [N, 2, 4]
        y = models.Normal(tf.tensordot(x, w, [[2], [0]]), 0.1, name='y')  # shape = [N, 2, 2]
```

**Comment by jcozar87** [comment]
Implemented in ba21dfc11ffbb41c18df07741ff20572ecf44b77
